core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -606,50 +569,16 @@ class HttpGoogleServicesDAO(val clientSecrets: GoogleClientSecrets,
     }
   }
 
-  override def testDMBillingAccountAccess(billingAccountName: RawlsBillingAccountName): Future[Boolean] = {
-    implicit val service = GoogleInstrumentedService.IamCredentials
-
-    /* Because we can't assume the identity of the Google SA that actually does the work in DM (it's in their project and we can't access it),
-       we've added billingprobe@terra-deployments-{env} to the Google Group we ask our users to add to their billing accounts.
-       In order to test that users have set up their billing accounts correctly, Rawls creates a token as the billingprobe@ SA and then
-       calls testIamPermissions as it to see if it has the scopes required to associate projects with billing accounts.
-       If it does, the group was correctly added, and the Google DM SA will be fine.
-       (This function would incorrectly return true if users were to add billingprobe@ to their project and not the group or DM SA, but we
-       don't publicise the existence of that account and it's not the thing we ask them to do, so the probability is near-zero.)
-
-       In order for all of this to work, Rawls needs iam.serviceAccountTokenCreator on either the terra-deployments-{env} project
-       or the billingprobe@ SA itself. We could also generate keys for the billingprobe@ SA and put them in Vault, but doing that and then
-       intermittently refreshing them is a chore.
-     */
-    // First, get an access token to act as the Google APIs Service Agent that Deployment Manager runs as.
-    val tokenRequestBody = new GenerateAccessTokenRequest().setScope(List(ComputeScopes.CLOUD_PLATFORM).asJava)
-    val saResourceName =
-      s"projects/-/serviceAccounts/$billingProbeEmail" // the dash is required; a project name will not work. https://bit.ly/2EXrXnj
-    val accessTokenRequest = getIAMCredentials(getDeploymentManagerAccountCredential)
-      .projects()
-      .serviceAccounts()
-      .generateAccessToken(saResourceName, tokenRequestBody)
-
-    for {
-      tokenResponse <- retryWhen500orGoogleError { () =>
-        blocking {
-          executeGoogleRequest(accessTokenRequest)
-        }
-      }
-
-      // Now we've got an access token, test IAM permissions to see if the SA has permission to create projects.
-      probeSACredential = buildCredentialFromAccessToken(tokenResponse.getAccessToken, billingProbeEmail)
-      hasAccess <- testBillingAccountAccess(billingAccountName, probeSACredential)
-    } yield hasAccess
-  }
+  override def testTerraBillingAccountAccess(billingAccountName: RawlsBillingAccountName): Future[Boolean] =
+    testBillingAccountAccess(billingAccountName, getBillingServiceAccountCredential)
```

Comment:
This is the only place where I changed functionality rather than just removed deprecated code. We used to check that this service account that is in `terra-billing@terra.bio` had permission on the user's billing account as a proxy for whether the user had granted access to `terra-billing@terra.bio`. I've replaced it with a check that our billing SA has access to the billing account. This is the SA that we use to update the billing account on workspace Google projects, so it's the account that actually needs access in order for workspace creation work.

Reply:
@marctalbott but it looks like `testBillingAccountAccess` itself calls `testTerraBillingAccountAccess`. How does this not cause infinite recursion?

Reply:
There's some slightly confusing overloading going on here.

The `testBillingAccountAccess` method that calls this takes a `UserInfo` along with the billing account name which then calls this method. This method calls a different `testBillingAccountAccess` that takes a `Credential` and issues the actual access check to GCP

Reply:
It'd be worth renaming one of them to aid readability and avoid any confusion. I can this being hard to debug 6 months from now.